charms/worker/k8s/src/charm.py

```diff
@@ -361,11 +369,19 @@ def _check_k8sd_ready(self):
         self.api_manager.check_k8sd_ready()
 
     def _get_extra_sans(self):
-        """Retrieve the certificate extra SANs."""
+        """Retrieve the certificate extra SANs.
+
+        Raises:
+            ReconcilerError: If the public address cannot be retrieved.
+        """
         extra_sans_str = str(self.config.get("kube-apiserver-extra-sans") or "")
-        configured_sans = {san for san in extra_sans_str.strip().split() if san}
-        all_sans = configured_sans | set([_get_public_address()])
-        return sorted(all_sans)
+        extra_sans = {san for san in extra_sans_str.strip().split() if san}
```

Comment:
Funny, I don't think pythons's `str.split()` can ever return an empty string.  I'm not sure if the filter checking is even necessary

```suggestion
        extra_sans = set(extra_sans_str.strip().split())
```